app/src/main/java/com/github/se/cyrcle/model/user/UserViewModel.kt

```diff
@@ -54,6 +54,25 @@ class UserViewModel(
         })
   }
 
+  /**
+   * Gets a user by ID from the Firestore database and calls the onSuccess callback with the user.
+   *
+   * @param userId the ID of the user to get
+   * @param onSuccess the callback to call with the user
+   */
+  fun getUserById(userId: String, onSuccess: (User) -> Unit) {
+    Log.d("UserViewModel", "Getting user by ID: $userId")
+    userRepository.getUserById(
+        userId,
+        onSuccess = { onSuccess(it) },
+        onFailure = { exception ->
+          Log.e(
+              "com.github.se.cyrcle.model.user.UserViewModel",
+              "Failed to fetch user by ID: $userId",
+              exception)
+        })
+  }
+
```

Comment:
I have seen no test for this new function. Would it be possible to have at least a small one ?

[Non-blocking note] Also, the tags for logging (`"UserViewModel"` and `"com.github.se.cyrcle.model.user.UserViewModel"`) don't match, maybe it's intended (as both are for different purposes)